code/modules/events/wizard/greentext.dm

```diff
@@ -75,21 +75,21 @@
 /obj/item/greentext/process()
 	if(last_holder && last_holder != new_holder) //Somehow it was swiped without ever getting dropped
 		to_chat(last_holder, span_warning("A sudden wave of failure washes over you..."))
-		last_holder.add_atom_colour("#FF0000", ADMIN_COLOUR_PRIORITY)
+		last_holder.add_atom_colour("#ff0000", ADMIN_COLOUR_PRIORITY)
 		last_holder = new_holder //long live the king
 
 /obj/item/greentext/Destroy(force)
 	LAZYREMOVE(SSticker.round_end_events, roundend_callback)
 	QDEL_NULL(roundend_callback) //This ought to free the callback datum, and prevent us from harddeling
-	for(var/mob/all_player_mobs as anything in GLOB.player_list)
+	for(var/mob/player as anything in GLOB.player_list)
 		var/message = "<span class='warning'>A dark temptation has passed from this world"
-		if(all_player_mobs in color_altered_mobs)
+		if(player in color_altered_mobs)
 			message += " and you're finally able to forgive yourself"
-			if(all_player_mobs.color == "#FF0000" || all_player_mobs.color == "#00FF00")
-				all_player_mobs.remove_atom_colour(ADMIN_COLOUR_PRIORITY)
+			if(player.color == "#ff0000" || player.color == "#00ff00")
+				player.remove_atom_colour(ADMIN_COLOUR_PRIORITY)
 		message += "...</span>"
 		if(!quiet)
-			to_chat(all_player_mobs, message)
+			to_chat(player, message)
```

Comment:
we can make the whole string concatenation procedure a lot less convoluted, just like this.

```dm
		var/list/messages = list(span_warning("A dark temptation has passed from this world!"))
		if(player in color_altered_mobs)
			messages += span_green("You're finally able to forgive yourself...")
			if(player.color == "#ff0000" || player.color == "#00ff00")
				player.remove_atom_colour(ADMIN_COLOUR_PRIORITY)
		if(!quiet)
			to_chat(player, messages.Join(" "))
```

Reworking this as a list of strings that gets `Join()`'d at the very end is (allegedly) a bit faster than doing base string concatenation, and it allows you that greater flexibility over the spans in the strings (which make it look a bit cooler imo). We can also use the span macros, much cleaner. This is optional since you didn't really touch this code but I wanted to mention this anyways.

Reply:
I used "\n" on the join because imo it looks a bit weird to have a color change mid line